ingestion-beam/README.md

```diff
@@ -448,6 +455,99 @@ echo '{"payload":"dGVzdA==","attributeMap":{"remote_addr":"63.245.208.195"}}' >
 "
 ```
 
+# Republisher Job
+
+A job for republishing subsets of decoded messages to new destinations.
+
+The primary intention is to produce smaller derived Pub/Sub topics so
+that consumers that only need a specific subset of messages don't incur
+the cost of reading the entire stream of decoded messages.
+
+The Republisher has the additional responsibility of marking messages as seen
+in `Cloud MemoryStore` for deduplication purposes. That functionality exists
+here to avoid the expense of an additional separate consumer of the full
+decoded topic.
+
+## Capabilities
+
+### Marking Messages As Seen
+
+The job needs to connect to Redis in order to mark `document_id`s of consumed
+messages as seen. The Decoder is able to use that information to drop duplicate
+messages flowing through the pipeline.
+
+### Debug Republishing
+
+If `--enableDebugDestination` is set, messages containing an `x_debug_id`
+attribute will be republished to a destination that's configurable at runtime.
+This is currently expected to be a feature specific to structured ingestion,
+so should not be set for `telemetry-decoded` input.
+
+### Per-`docType` Republishing
+
+If `--perDocTypeEnabledList` is provided, a separate producer will be created
+for each `docType` specified in the given comma-separated list.
+See the `--help` output for details on format.
+
+### Per-Channel Sampled Republishing
+
+If `--perChannelSampleRatios` is provided, a separate producer will be created
```

Comment:
What happens to messages with values of `channel` outside the listed set?

Reply:
If a channel is not explicitly listed, then there is no output topic for that channel and no messages of that channel type are republished via this perChannel mechanism.